src/crab/ebpf_domain.cpp

```diff
@@ -2154,8 +2154,61 @@ void ebpf_domain_t::do_mem_store(const Mem& b, Type val_type, SValue val_svalue,
     });
 }
 
-void ebpf_domain_t::operator()(const LockAdd& a) {
-    // nothing to do here
+void ebpf_domain_t::operator()(const Atomic& a) {
+    if (m_inv.is_bottom())
+        return;
+    if (!m_inv.entail(type_is_pointer(reg_pack(a.access.basereg))) ||
+        !m_inv.entail(type_is_number(reg_pack(a.valreg)))) {
+        return;
+    }
+
+    // Fetch the current value into the R11 pseudo-register.
+    Reg r11{R11_ATOMIC_SCRATCH};
+    Mem mem = {.access = a.access, .value = r11, .is_load = true};
+    (*this)(mem);
```

Comment:
I think it's nicer to create a temporary instead, here and below, so the reader won't need to remember what was mem initialized to:
```suggestion
    (*this)(Mem{.access = a.access, .value = r11, .is_load = true});
```

Reply:
I apologize, I meant the suggestion as an example, not as something to commit directly. Sorry!

Reply:
Fixed